examples/mir_demo_server/server_example.cpp

```diff
@@ -150,7 +150,9 @@ try
         me::add_glog_options_to,
         miral::StartupInternalClient{spinner},
         miral::X11Support{},
-        miral::WaylandExtensions{"wl_shell:xdg_wm_base:zxdg_shell_v6:zwlr_layer_shell_v1"},
+        with_extension(miral::WaylandExtensions{"wl_shell:xdg_wm_base:zxdg_shell_v6:zwlr_layer_shell_v1"},
+            "xdg_decoration_unstable_v1", [](wl_display*) { return std::shared_ptr<void>{}; },
+            "some_other_extension", [](wl_display*) { return std::shared_ptr<void>{}; }),
```

Comment:
This seems unnecessarily complicated on both sides of the API. Does it make sense to combine the default extensions and the custom extensions in the same function? Would it make more sense to ask the user for a single `WaylandExtensionBuilder`, which creates all the extensions once a `wl_display` is provided?

Reply:
I don't see why we need another "Wayland Extensions" object (_Frustra fit per plura quod potest fieri per pauciora)_.

I rejected using a member function...
```c++
        miral::WaylandExtensions{"wl_shell:xdg_wm_base:zxdg_shell_v6:zwlr_layer_shell_v1"}
            .with_extension("xdg_decoration_unstable_v1", [](wl_display*) { return std::shared_ptr<void>{}; })
            .with_extension("some_other_extension", [](wl_display*) { return std::shared_ptr<void>{}; }),
```
... as it seems counter-intuitive to make `with_extension()` a `const` member function.

Reply:
Also, I had been thinking we simply add these bespoke extensions to the `wayland_extensions` option which gives the shell the opportunity to default them on (and also allows that choice to be overridden). That logic all belongs in `WaylandExtensions`.

That brings up an additional point: the shell ought have a way to validate the actual choice extensions being offered.